Backend/ManagerLayer/Login/LoginManager.cs

```diff
@@ -58,13 +59,17 @@ public bool LoginCheckUserDisabled()
             }
         }
 
-        public bool LoginCheckPassword(string password)
+        public bool LoginCheckPassword(LoginRequest request)
         {
             using (var _db = new DatabaseContext())
             {
-                string hashedPassword = _passwordService.HashPassword(password, user.PasswordSalt);
+                user = _userService.GetUser(_db, request.email);
+                string hashedPassword = _passwordService.HashPassword(request.password, user.PasswordSalt);
                 if (userRepo.ValidatePassword(user, hashedPassword))
                 {
+                    user.IncorrectPasswordCount = 0;
+                    _userService.UpdateUser(_db, user);
+                    _db.SaveChanges();
```

Comment:
When calling .savechanges(), catch any errors that can be thrown, and when caught, rollback changes, you can look at register route.